server/bleep/src/webserver/answer.rs

```diff
@@ -1251,22 +1250,20 @@ impl Agent {
         while let Some(fragment) = stream.next().await {
             let fragment = fragment?;
             response += &fragment;
-            self.update(Update::Article(fragment)).await?;
+
+            if let Some((article, summary)) = split_article_summary(&response) {
+                self.update(Update::Article(article)).await?;
+                self.update(Update::Conclude(summary)).await?;
+            } else {
+                self.update(Update::Article(response.clone())).await?;
+            }
         }
 
-        let article_conclusion_messages = vec![
-            "I hope that was useful, can I help with anything else?",
-            "Is there anything else I can help you with?",
-            "Can I help you with anything else?",
-        ];
+        let summary = split_article_summary(&response)
```

Comment:
Maybe we should backoff to a canned answer if the model doesn't generate a summary?

Reply:
Added as a fallback in 3cd6c943453329a32f078c5daa4390dd3d7860b8